### PR TITLE
LPS-158088 Allow validations for System Objects

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -181,6 +182,58 @@ public ObjectValidationRule updateObjectValidationRule(
 		return objectValidationRulePersistence.update(objectValidationRule);
 	}
 
+	@Override
+	public void validate(
+			Object object, long objectDefinitionId,
+			Map<String, Object> modelAttributes)
+		throws PortalException {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:189]: Hi @seltong,

try to merge this method with the other one for ObjectEntry by using  `instanceof`.

Thanks!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java [hunk 2]
```diff
@@ -48,7 +49,9 @@ public static List<Map<String, Object>> getCodeEditorElements(
 					_objectFieldLocalService.getObjectFields(
 						objectDefinitionId),
 					objectField -> HashMapBuilder.put(
-						"content", objectField.getName()
+						"content",
+						StringUtil.removeSubstring(
+							objectField.getDBColumnName(), StringPool.UNDERLINE)
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java:54]: Hi @seltong,

Slack me tomorrow to talk about this approach, I believe we can try to keep the objectFieldName for Custom Objects and get the DBColumnName for System ones.

Thanks!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/frontend/taglib/servlet/taglib/ObjectDefinitionsValidationsScreeNavigationCategory.java [hunk 3]
```diff
@@ -73,7 +75,10 @@ public String getLabel(Locale locale) {
 
 	@Override
 	public boolean isVisible(User user, ObjectDefinition objectDefinition) {
-		if (objectDefinition.isDefaultStorageType()) {
+		if ((GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-151671")) ||
+			 !objectDefinition.isSystem()) &&
+			objectDefinition.isDefaultStorageType()) {
+
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/frontend/taglib/servlet/taglib/ObjectDefinitionsValidationsScreeNavigationCategory.java:81]: Hi @seltong,

I believe we can invert this conditionals, I mean something like this:

````
if (!objectDefinition.isDefaultStorageType()) {
  return false;
}

if (!objectDefinition.isSystem()) {
   return true;
}

return GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-151671");
````

**Comment by seltong** [comment]
ci:test:sf

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:sf

**Pull request closed without merging**